scalameta/transversers/shared/src/main/scala/scala/meta/transversers/SimpleTraverser.scala

```diff
@@ -0,0 +1,6 @@
+package scala.meta
+package transversers
+
+private[meta] class SimpleTraverser {
+  def apply(tree: Tree): Unit = tree.children.foreach(apply)
```

Comment:
Neither traversers are stack safe. It turns out that there is a file in the Kafka repo that causes the existing traverser to blow up the stack (bug?): https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/KafkaConfig.scala:
```
Exception in thread "main" java.lang.StackOverflowError
	at scala.meta.Term$Apply$TermApplyImpl.args(Trees.scala:81)
	at scala.meta.Term$Apply$TermApplyImpl.args(Trees.scala:81)
	at scala.meta.Term$Apply$TermApplyImpl.args(Trees.scala:81)
```

I also benchmarked a stack-safe traverser but turned out to be 30-50% slower than this one.

Reply:
It's normal that users need to increase the stack size to use scalameta, we probably don't do a good enough job at documenting it however 🤔